diff_cover/report_generator.py

```diff
@@ -172,15 +172,29 @@ def _diff_violations(self):
 
         To make this efficient, we cache and reuse the result.
         """
+        src_paths_changed = self._diff.src_paths_changed()
         if not self._diff_violations_dict:
-            self._diff_violations_dict = {
-                src_path: DiffViolations(
-                    self._violations.violations(src_path),
-                    self._violations.measured_lines(src_path),
-                    self._diff.lines_changed(src_path),
+            try:
+                violations = self._violations.violations_batch(
```

Comment:
This code is very similar to the `except` block, but it uses results from `violations_batch()` rather than calling `violations()`.

It duplicates some code from the `except` block, but overall the duplication seems to make the code more readable by keeping the two implementations (using `violations_batch()` versus `violations()`) distinct from one another.